linked_list/implementation.py

```diff
@@ -1,40 +1,127 @@
 from .interface import AbstractLinkedList
-
+from .node import Node
 
 class LinkedList(AbstractLinkedList):
     """
     Implementation of an AbstractLinkedList inteface.
     """
 
     def __init__(self, elements=None):
-        pass
+        self.start = None
+        self.end = None
+        if elements:
+            for e in elements:
+                node = Node(e)
+                if not self.start:
+                    self.start = node
+                if self.end:
+                    self.end.next = node
+                self.end = node
 
     def __str__(self):
-        pass
+        str_list = []
```

Comment:
This is just for the string representation of an instance of a class. The only reason a list is being used here is because it formats nicer when being output as a string. (Tests are expecting a string output of '[element, element, element]' The list here is being used as something tidier than assembling a string. If we're avoiding the use of a list we would be doing something along the lines of:
```
if len(self) = 0:
    return "[]"
else:
    string_representation = "[
    for i in iter(self):
        string_representation += str(i)
        string_representation += ", "
    #remove trailing comma and space
    string_representation = string_representation[:-2]
    string_representation += ]
    return string_representation
```